### DISCUSSION TITLE
[Proposal] Custom Async Enumerable types

### DISCUSSION THREAD

**Discussion by Khitiara** [dotnet/csharplang#6189] [Language Ideas] [answered]
Proposal is to add a similar interface for asynchronous iterator methods as exists using AsyncMethodBuilderAttribute for custom Task-like types

My current use case is a project similar to https://github.com/Cysharp/UniTask where custom task-like types are defined which use the game engine framework for continuations, and being able to define a custom `IAsyncEnumerable`-alike type and use yield for it would be convenient

**Comment by timcassell** [comment, marked as answer]
#3629
